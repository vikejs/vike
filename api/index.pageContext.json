{"_urlRewrite":null,"pageId":"/pages/api","routeParams":{}}