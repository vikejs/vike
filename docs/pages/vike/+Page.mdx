See [vite-plugin-ssr.com/vike](https://vite-plugin-ssr.com/vike).
