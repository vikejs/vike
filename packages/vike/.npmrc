# Throw an error at install-time if user has a Node.js version lower than package.json#engines.node
engine-strict=true
