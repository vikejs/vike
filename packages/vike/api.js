// Some tools still need this as of January 2025
export * from './dist/esm/node/api/index.js'
export { default } from './dist/esm/node/api/index.js'
