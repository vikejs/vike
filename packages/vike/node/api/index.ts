export { build_ as build }
export { dev_ as dev }
export { preview_ as preview }
export { prerender_ as prerender }

import type { build } from './build.js'
import type { dev } from './dev.js'
import type { preview } from './preview.js'
import type { prerender } from './prerender.js'
import { assertIsNotBrowser } from '../../utils/assertIsNotBrowser.js'
assertIsNotBrowser()

const build_: typeof build = async (...args) => (await import('./build.js')).build(...args)
const dev_: typeof dev = async (...args) => (await import('./dev.js')).dev(...args)
const preview_: typeof preview = async (...args) => (await import('./preview.js')).preview(...args)
const prerender_: typeof prerender = async (...args) => (await import('./prerender.js')).prerender(...args)
