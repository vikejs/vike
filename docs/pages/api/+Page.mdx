import { Link, Warning } from '@brillout/docpress'

Programmatically trigger <Link href="/cli">CLI commands</Link>.

```js
import { dev, build, preview, prerender } from 'vike/api'

// Same as `$ vike dev`
await dev()
console.log('Development server is ready')

// Same as `$ vike build`
await build()
console.log('Build is done')

// Same as `$ vike preview`
await preview()
console.log('Preview server is ready')

// Same as `$ vike prerender`
await prerender()
console.log('Pre-rendering is done')
```

Common use cases:
 - Building your own CLI in the context of <Link href="/build-your-own-framework" noBreadcrumb />.
 - Writing a custom build script, for example to add a post-processing build step.


## `dev()`

Start the development server.

```js
import { dev } from 'vike/api'
const {
  viteConfig, // Resolved Vite configuration
  viteServer, // Vite's development server
} = await dev({
  viteConfig // Vite configuration (optional)
})
console.log('Development server is ready')
```

See: <Link href='#viteconfig'>`viteConfig` option</Link>.

Vite types:
- [ResolvedConfig](https://vite.dev/guide/api-javascript.html#resolvedconfig) types the `viteConfig` return value.
- Vite's development server: [`const viteServer = createServer()`](https://vite.dev/guide/api-javascript.html#createserver).

## `build()`

Build for production.

```js
import { build } from 'vike/api'
const {
  rollupOutputClient, // Generated dist/client/ files for the client-side
  rollupOutputServer, // Generated dist/server/ files for the server-side
} = await build({
  viteConfig // Vite configuration (optional)
})
console.log('Build is done')
```

See: <Link href='#viteconfig'>`viteConfig` option</Link>.


## `preview()`

Start preview server using production build (only works for <Link href="/glossary#ssg">SSG apps</Link>).

```js
import { preview } from 'vike/api'
const {
  viteConfig, // Resolved Vite configuration
  viteServer, // Vite's development server
} = await preview({
  viteConfig // Vite configuration (optional)
})
console.log('Preview server is ready')
```

See: <Link href='#viteconfig'>`viteConfig` option</Link>.

Vite types:
- [ResolvedConfig](https://vite.dev/guide/api-javascript.html#resolvedconfig) types the `viteConfig` return value.
- Vite's preview server: [`const viteServer = preview()`](https://vite.dev/guide/api-javascript.html#preview).


## `prerender()`

<Link href="/pre-rendering">Pre-render</Link> pages (only needed when <Link href="/prerender#disableautorun">`prerender.disableAutoRun`</Link> is `true`).

```js
import { prerender } from 'vike/api'
const {
  viteConfig // Resolved Vite configuration
} = await prerender({
  viteConfig, // Vite configuration (optional)
  pageContextInit, // Initial pageContext (optional)
  onPagePrerender // Called before writing an HTML file (optional)
})
console.log('Pre-rendering is done')
```

For more settings, see <Link href="/prerender" />.

Vite types:
- [ResolvedConfig](https://vite.dev/guide/api-javascript.html#resolvedconfig) types the `viteConfig` return value.

Options:
- **`viteConfig`**: See <Link href='#viteconfig'>`viteConfig` option</Link>.
- **`pageContextInit`**: Initial <Link href="/pageContext">`pageContext`</Link> values.
  ```js
  prerender({
    pageContextInit: {
      someData: 42,
      // ...
    }
  })
  ```
  See also: <Link href="/pageContext#lifecycle" />.
- **`onPagePrerender()`**: control where/how HTML files are written.
  <Warning>Don't use this option without having contacted a Vike maintainer: this functionality may be changed/removed at any time.</Warning>
  ```js
  prerender({
    // If onPagePrerender() is set, then Vike won't create HTML files
    onPagePrerender(pageContext) {
      // Cusotm logic for writing HTML files to the filesystem
      // ...
    }
  })
  ```


## `viteConfig`

All API functions accept the option `viteConfig` (with type [`InlineConfig`](https://vitejs.dev/guide/api-javascript.html#inlineconfig)).

> We generally recommend to define your [Vite settings](https://vite.dev/config/) in your `vite.config.js` file instead of using the `viteConfig` option. The API automatically loads your `vite.config.js` file (just like the CLI).

If you want to define Vite settings outside of your app (typically when <Link href="/build-your-own-framework">building your own framework</Link>) you can do this:

```js
import { dev } from 'vike/api'

await dev({
  viteConfig: {
    // vite.config.js can live in node_modules/my-framework/src/vite.config.ts
    configFile: './path/to/vite.config.js',
    // The app's root can be somewhere completely else than vite.config.js
    root: './path/to/app/'
    // Some Vite settings overriding vite.config.js
    /* ... */
  }
})
console.log('Development server is ready')
```

If you want to define your entire Vite config programmatically:

```js
import { build } from 'vike/api'

await build({
  viteConfig: {
    // Don't load any vite.config.js
    configFile: false,
    // All Vite settings
    /* ... */
  }
})
console.log('Build is done')
```


## See also

- [Vite > JavaScript API](https://vite.dev/guide/api-javascript)
- <Link href="/cli" />
- <Link href="/build-your-own-framework" />
- <Link href="/createDevMiddleware" />
