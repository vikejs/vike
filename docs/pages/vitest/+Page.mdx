import { Link } from '@brillout/docpress'

When running Vitest consider either:
 - Using Vike's JavaScript API, see <Link href="#with-vike" />.
 - Removing Vike's Vite plugin, see <Link href="#without-vike" />.

## With Vike

If you want to use Vitest to test your Vike app, then you can use <Link href="/api">Vike's JavaScript API</Link>:

```ts
// dev.spec.ts

import { afterAll, beforeAll } from 'vitest'
import { testApp, viteConfig } from './testApp'
import { dev } from 'vike/api'

let viteServer: Awaited<ReturnType<typeof dev>>['viteServer']

beforeAll(async () => {
  const { viteServer } = await dev({ viteConfig })
  await viteServer.listen()
  await sleep(10) // avoid race condition of server not actually being ready
}, 10 * 1000)

afterAll(async () => {
  await viteServer.close()
})

testApp()

function sleep(milliseconds: number): Promise<void> {
  return new Promise((r) => setTimeout(r, milliseconds))
}
```

```ts
// preview.spec.ts

import { afterAll, beforeAll } from 'vitest'
import { testApp, viteConfig } from './testApp'
import { build, preview } from 'vike/api'

let viteServer: Awaited<ReturnType<typeof preview>>['viteServer']

beforeAll(async () => {
  await build({ viteConfig })
  const { viteServer } = await preview({ viteConfig })
  // If you want to replicate Vite's startup log:
  viteServer!.printUrls()
  // For the preview server, no need to call server.listen()
}, 40 * 1000)

afterAll(async () => {
  await viteServer!.close()
})

testApp()
```

```ts
// testApp.ts

export { testApp }
export { viteConfig }

import { expect, describe, it } from 'vitest'

const viteConfig = {
  logLevel: 'warn' as const,
  root: __dirname,
  configFile: __dirname + '/vite.config.js'
}
const urlBase = 'http://localhost:3000'

function testApp() {
  describe('Vitest', () => {
    it('run Vitest with Vike', { timeout: 10 * 1000 }, async () => {
      {
        const html = await fetchHtml('/')
        expect(html).toContain('<h1>Welcome</h1>')
        expect(html).toContain('<li>Rendered to HTML.</li>')
      }
      {
        const html = await fetchHtml('/about')
        expect(html).toContain('<h1>About</h1>')
        expect(html).toContain('<p>Example of using Vike.</p>')
      }
    })
  })
}

async function fetchHtml(urlPathname: string) {
  const ret = await fetch(urlBase + urlPathname)
  const html = await ret.text()
  return html
}
```

> Feel free to [reach out on GitHub Discussions](https://github.com/vikejs/vike/discussions) if you want help setting up Vitest with Vike differently.


## Without Vike

If you want to test JavaScript code without testing your Vike app, then remove <Link href="/vite-plugin">Vike's Vite plugin</Link> when running Vitest:

```ts
// vite.config.ts

import type { UserConfig } from 'vite'
import vike from 'vike/plugin'

export default {
  plugin: [
    // Always add Vike's Vite plugin// [!code --]
    vike(),// [!code --]
    // Don't add Vike's Vite plugin when running Vitest// [!code ++]
    !process.env.VITEST && vike(),// [!code ++]
    // ...
  ]
} satisfies UserConfig
```

So that you can share your `vite.config.js` settings between Vitest and your Vike app.
