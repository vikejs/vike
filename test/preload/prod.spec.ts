import { beforeAll } from 'vitest'
import { build } from 'vike/api'
import { testRun } from './testRun'

beforeAll(async () => {
  await buildApp()
}, 40 * 1000)

testRun(false)

async function buildApp() {
  await build({
    viteConfig: {
      logLevel: 'warn' as const,
      root: __dirname,
      configFile: __dirname + '/vite.config.js',
    },
  })
}
