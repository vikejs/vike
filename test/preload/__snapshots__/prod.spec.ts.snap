// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`preload tags > Preload Default 1`] = `
[
  {
    "assetType": "style",
    "earlyHintLink": "</assets/static/renderer_Layout-031b266d.$HASH.css>; rel=preload; as=style",
    "isEntry": true,
    "mediaType": "text/css",
    "src": "/assets/static/renderer_Layout-031b266d.$HASH.css",
  },
  {
    "assetType": "font",
    "earlyHintLink": "</assets/static/Sono-Light.$HASH.woff2>; rel=preload; as=font",
    "isEntry": false,
    "mediaType": "font/woff2",
    "src": "/assets/static/Sono-Light.$HASH.woff2",
  },
  {
    "assetType": "image",
    "earlyHintLink": "</assets/static/logo.$HASH.svg>; rel=preload; as=image",
    "isEntry": false,
    "mediaType": "image/svg+xml",
    "src": "/assets/static/logo.$HASH.svg",
  },
  {
    "assetType": "script",
    "earlyHintLink": "</assets/entries/entry-server-routing.$HASH.js>; rel=modulepreload; as=script",
    "isEntry": true,
    "mediaType": "text/javascript",
    "src": "/assets/entries/entry-server-routing.$HASH.js",
  },
  {
    "assetType": "script",
    "earlyHintLink": "</assets/entries/pages_index.$HASH.js>; rel=modulepreload; as=script",
    "isEntry": false,
    "mediaType": "text/javascript",
    "src": "/assets/entries/pages_index.$HASH.js",
  },
  {
    "assetType": "script",
    "earlyHintLink": "</assets/chunks/chunk-$HASH.js>; rel=modulepreload; as=script",
    "isEntry": false,
    "mediaType": "text/javascript",
    "src": "/assets/chunks/chunk-$HASH.js",
  },
]
`;

exports[`preload tags > Preload Default 2`] = `
"<!DOCTYPE html>
    <html>
      <head>
        <link rel="stylesheet" type="text/css" href="/assets/static/renderer_Layout-031b266d.$HASH.css">
        <link rel="preload" href="/assets/static/Sono-Light.$HASH.woff2" as="font" type="font/woff2" crossorigin>
      </head>
      <body>
        <div id="root"><div style="display:flex;max-width:900px;margin:auto"><div style="padding:20px;padding-top:20px;flex-shrink:0;display:flex;flex-direction:column;align-items:center;line-height:1.8em"><div style="margin-top:20px;margin-bottom:10px"><a href="/"><img src="/assets/static/logo.$HASH.svg" height="64" width="64"/></a></div><a class="navitem" href="/">Preload Default</a><a class="navitem" href="/preload-disabled">Preload Disabled</a><a class="navitem" href="/preload-images">Preload Images</a></div><div style="padding:20px;padding-bottom:50px;border-left:2px solid #eee;min-height:100vh"><h1>Default</h1><p>This page showcases the default preloading strategy: in production, both the image and the font are preloaded.</p></div></div></div>
        <script id="vike_pageContext" type="application/json">{"pageProps":"!undefined","abortReason":"!undefined","_urlRewrite":null,"_urlRedirect":"!undefined","abortStatusCode":"!undefined","_abortCall":"!undefined","_pageContextInitIsPassedToClient":"!undefined","pageId":"/pages/index","routeParams":{},"data":"!undefined"}</script>
        <script src="/assets/entries/entry-server-routing.$HASH.js" type="module" async></script>
        <link rel="modulepreload" href="/assets/entries/pages_index.$HASH.js" as="script" type="text/javascript">
        <link rel="modulepreload" href="/assets/chunks/chunk-$HASH.js" as="script" type="text/javascript">
      </body>
    </html>"
`;

exports[`preload tags > Preload Disabled 1`] = `
[
  {
    "assetType": "style",
    "earlyHintLink": "</assets/static/renderer_Layout-031b266d.$HASH.css>; rel=preload; as=style",
    "isEntry": true,
    "mediaType": "text/css",
    "src": "/assets/static/renderer_Layout-031b266d.$HASH.css",
  },
  {
    "assetType": "font",
    "earlyHintLink": "</assets/static/Sono-Light.$HASH.woff2>; rel=preload; as=font",
    "isEntry": false,
    "mediaType": "font/woff2",
    "src": "/assets/static/Sono-Light.$HASH.woff2",
  },
  {
    "assetType": "image",
    "earlyHintLink": "</assets/static/logo.$HASH.svg>; rel=preload; as=image",
    "isEntry": false,
    "mediaType": "image/svg+xml",
    "src": "/assets/static/logo.$HASH.svg",
  },
  {
    "assetType": "script",
    "earlyHintLink": "</assets/entries/entry-server-routing.$HASH.js>; rel=modulepreload; as=script",
    "isEntry": true,
    "mediaType": "text/javascript",
    "src": "/assets/entries/entry-server-routing.$HASH.js",
  },
  {
    "assetType": "script",
    "earlyHintLink": "</assets/entries/pages_preload-disabled.$HASH.js>; rel=modulepreload; as=script",
    "isEntry": false,
    "mediaType": "text/javascript",
    "src": "/assets/entries/pages_preload-disabled.$HASH.js",
  },
  {
    "assetType": "script",
    "earlyHintLink": "</assets/chunks/chunk-$HASH.js>; rel=modulepreload; as=script",
    "isEntry": false,
    "mediaType": "text/javascript",
    "src": "/assets/chunks/chunk-$HASH.js",
  },
]
`;

exports[`preload tags > Preload Disabled 2`] = `
"<!DOCTYPE html>
    <html>
      <head>
        <link rel="stylesheet" type="text/css" href="/assets/static/renderer_Layout-031b266d.$HASH.css">
      </head>
      <body>
        <div id="root"><div style="display:flex;max-width:900px;margin:auto"><div style="padding:20px;padding-top:20px;flex-shrink:0;display:flex;flex-direction:column;align-items:center;line-height:1.8em"><div style="margin-top:20px;margin-bottom:10px"><a href="/"><img src="/assets/static/logo.$HASH.svg" height="64" width="64"/></a></div><a class="navitem" href="/">Preload Default</a><a class="navitem" href="/preload-disabled">Preload Disabled</a><a class="navitem" href="/preload-images">Preload Images</a></div><div style="padding:20px;padding-bottom:50px;border-left:2px solid #eee;min-height:100vh"><h1>Preload Disabled</h1><p>This page showcases completely disabled preloading: the image nor the font are preloaded.</p></div></div></div>
        <script id="vike_pageContext" type="application/json">{"pageProps":"!undefined","abortReason":"!undefined","_urlRewrite":null,"_urlRedirect":"!undefined","abortStatusCode":"!undefined","_abortCall":"!undefined","_pageContextInitIsPassedToClient":"!undefined","pageId":"/pages/preload-disabled","routeParams":{},"data":"!undefined"}</script>
        <script src="/assets/entries/entry-server-routing.$HASH.js" type="module" async></script>
      </body>
    </html>"
`;

exports[`preload tags > Preload Eager 1`] = `
[
  {
    "assetType": "style",
    "earlyHintLink": "</assets/static/renderer_Layout-031b266d.$HASH.css>; rel=preload; as=style",
    "isEntry": true,
    "mediaType": "text/css",
    "src": "/assets/static/renderer_Layout-031b266d.$HASH.css",
  },
  {
    "assetType": "font",
    "earlyHintLink": "</assets/static/Sono-Light.$HASH.woff2>; rel=preload; as=font",
    "isEntry": false,
    "mediaType": "font/woff2",
    "src": "/assets/static/Sono-Light.$HASH.woff2",
  },
  {
    "assetType": "image",
    "earlyHintLink": "</assets/static/logo.$HASH.svg>; rel=preload; as=image",
    "isEntry": false,
    "mediaType": "image/svg+xml",
    "src": "/assets/static/logo.$HASH.svg",
  },
  {
    "assetType": "script",
    "earlyHintLink": "</assets/entries/entry-server-routing.$HASH.js>; rel=modulepreload; as=script",
    "isEntry": true,
    "mediaType": "text/javascript",
    "src": "/assets/entries/entry-server-routing.$HASH.js",
  },
  {
    "assetType": "script",
    "earlyHintLink": "</assets/entries/pages_preload-eager.$HASH.js>; rel=modulepreload; as=script",
    "isEntry": false,
    "mediaType": "text/javascript",
    "src": "/assets/entries/pages_preload-eager.$HASH.js",
  },
  {
    "assetType": "script",
    "earlyHintLink": "</assets/chunks/chunk-$HASH.js>; rel=modulepreload; as=script",
    "isEntry": false,
    "mediaType": "text/javascript",
    "src": "/assets/chunks/chunk-$HASH.js",
  },
]
`;

exports[`preload tags > Preload Eager 2`] = `
"<!DOCTYPE html>
    <html>
      <head>
        <link rel="stylesheet" type="text/css" href="/assets/static/renderer_Layout-031b266d.$HASH.css">
        <link rel="preload" href="/assets/static/Sono-Light.$HASH.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/assets/static/Sono-Light.$HASH.ttf" as="font" type="font/ttf" crossorigin>
        <link rel="preload" href="/assets/static/logo.$HASH.svg" as="image" type="image/svg+xml">
        <link rel="modulepreload" href="/assets/entries/pages_preload-eager.$HASH.js" as="script" type="text/javascript">
        <link rel="modulepreload" href="/assets/chunks/chunk-$HASH.js" as="script" type="text/javascript">
      </head>
      <body>
        <div id="root"><div style="display:flex;max-width:900px;margin:auto"><div style="padding:20px;padding-top:20px;flex-shrink:0;display:flex;flex-direction:column;align-items:center;line-height:1.8em"><div style="margin-top:20px;margin-bottom:10px"><a href="/"><img src="/assets/static/logo.$HASH.svg" height="64" width="64"/></a></div><a class="navitem" href="/">Preload Default</a><a class="navitem" href="/preload-disabled">Preload Disabled</a><a class="navitem" href="/preload-images">Preload Images</a></div><div style="padding:20px;padding-bottom:50px;border-left:2px solid #eee;min-height:100vh"><h1>Eager</h1><p>This page showcases eager preloading (non-JavaScript assets are preloaded ASAP).</p></div></div></div>
        <script id="vike_pageContext" type="application/json">{"pageProps":"!undefined","abortReason":"!undefined","_urlRewrite":null,"_urlRedirect":"!undefined","abortStatusCode":"!undefined","_abortCall":"!undefined","_pageContextInitIsPassedToClient":"!undefined","pageId":"/pages/preload-eager","routeParams":{},"data":"!undefined"}</script>
        <script src="/assets/entries/entry-server-routing.$HASH.js" type="module" async></script>
      </body>
    </html>"
`;

exports[`preload tags > Preload Images 1`] = `
[
  {
    "assetType": "style",
    "earlyHintLink": "</assets/static/renderer_Layout-031b266d.$HASH.css>; rel=preload; as=style",
    "isEntry": true,
    "mediaType": "text/css",
    "src": "/assets/static/renderer_Layout-031b266d.$HASH.css",
  },
  {
    "assetType": "font",
    "earlyHintLink": "</assets/static/Sono-Light.$HASH.woff2>; rel=preload; as=font",
    "isEntry": false,
    "mediaType": "font/woff2",
    "src": "/assets/static/Sono-Light.$HASH.woff2",
  },
  {
    "assetType": "image",
    "earlyHintLink": "</assets/static/logo.$HASH.svg>; rel=preload; as=image",
    "isEntry": false,
    "mediaType": "image/svg+xml",
    "src": "/assets/static/logo.$HASH.svg",
  },
  {
    "assetType": "script",
    "earlyHintLink": "</assets/entries/entry-server-routing.$HASH.js>; rel=modulepreload; as=script",
    "isEntry": true,
    "mediaType": "text/javascript",
    "src": "/assets/entries/entry-server-routing.$HASH.js",
  },
  {
    "assetType": "script",
    "earlyHintLink": "</assets/entries/pages_preload-images.$HASH.js>; rel=modulepreload; as=script",
    "isEntry": false,
    "mediaType": "text/javascript",
    "src": "/assets/entries/pages_preload-images.$HASH.js",
  },
  {
    "assetType": "script",
    "earlyHintLink": "</assets/chunks/chunk-$HASH.js>; rel=modulepreload; as=script",
    "isEntry": false,
    "mediaType": "text/javascript",
    "src": "/assets/chunks/chunk-$HASH.js",
  },
]
`;

exports[`preload tags > Preload Images 2`] = `
"<!DOCTYPE html>
    <html>
      <head>
        <link rel="stylesheet" type="text/css" href="/assets/static/renderer_Layout-031b266d.$HASH.css">
        <link rel="preload" href="/assets/static/Sono-Light.$HASH.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/assets/static/logo.$HASH.svg" as="image" type="image/svg+xml">
      </head>
      <body>
        <div id="root"><div style="display:flex;max-width:900px;margin:auto"><div style="padding:20px;padding-top:20px;flex-shrink:0;display:flex;flex-direction:column;align-items:center;line-height:1.8em"><div style="margin-top:20px;margin-bottom:10px"><a href="/"><img src="/assets/static/logo.$HASH.svg" height="64" width="64"/></a></div><a class="navitem" href="/">Preload Default</a><a class="navitem" href="/preload-disabled">Preload Disabled</a><a class="navitem" href="/preload-images">Preload Images</a></div><div style="padding:20px;padding-bottom:50px;border-left:2px solid #eee;min-height:100vh"><h1>Preload Images</h1><p>This page showcases a custom strategy of preloading images.</p></div></div></div>
        <script id="vike_pageContext" type="application/json">{"pageProps":"!undefined","abortReason":"!undefined","_urlRewrite":null,"_urlRedirect":"!undefined","abortStatusCode":"!undefined","_abortCall":"!undefined","_pageContextInitIsPassedToClient":"!undefined","pageId":"/pages/preload-images","routeParams":{},"data":"!undefined"}</script>
        <script src="/assets/entries/entry-server-routing.$HASH.js" type="module" async></script>
        <link rel="modulepreload" href="/assets/entries/pages_preload-images.$HASH.js" as="script" type="text/javascript">
        <link rel="modulepreload" href="/assets/chunks/chunk-$HASH.js" as="script" type="text/javascript">
      </body>
    </html>"
`;
